tensorrt_llm/_torch/modules/linear.py

```diff
@@ -614,7 +614,7 @@ def apply(self, module: Linear, input: torch.Tensor,
         assert input.dtype == torch.bfloat16
 
         if get_sm_version() == 100:
-            if module.use_cute_dsl_blockscaling_mm:
+            if module.use_cute_dsl_blockscaling_mm or module.disable_deep_gemm:
```

Comment:
How many GEMM kernels do we currently have?

I suggest we replace the boolean flags used for kernel selection (e.g., `use_cute_dsl_blockscaling_mm`, `disable_deep_gemm`) with a single enum. This would make the code cleaner, safer, and easier to extend